Installer/bin/libAprilTags.sh

```
@@ -19,6 +19,47 @@ function getProcPID() {
     echo 0
 }
 
+echoerr() { echo "$@" 1>&2; } # bypass capture by var assigning because std::err
+
 if [[ $1 == "pid" ]]; then
     getProcPID $2
+elif [[ $1 == "camIDs" ]]; then
+    cd "${0%/*}"/camerascanner
+    # run the go file in the most efficent way possible
+    arch=$(uname -m)
+    if [[ -f "scanner_${arch}" ]]; then
+        ret=$("./scanner_${arch}")
+    else
+        echoerr "running go file directly"
+        ret=$(go run main)
+    fi
+    
+    if [[ "${ret,,}" == *"err"* ]] || [[ "${ret,,}" == *"fault"* ]]; then
+    	echoerr "there was an error..."
```

Can this be more descriptive?

---

error should be displayed, too
this probably hides the error output

---

Just a note, the go code has no error handling, so this is basically just taking care of ensuring that the parameters returned are not malformed due to some odd novel error that I have not seen yet. (this also applies to your other comment below).

---

@cwosw the ask is to return an error message that is self-descriptive, e.g. "Error encountered parsing the camera serial numbers from /dev/vrl/by-ids/" or something of that nature.  When debugging, and we see an error message "there was an error..." written to a log, it isn't descriptive enough to help us debug what is going on.  Can you make the error messages more self descriptive in this file please?

---

The issue is that the majority of errors that could be coming from this are issues with `go build main`. I can not control those, and they already get thrown to console. I saw no need to make it more verbose if the errors that were important were already written to console by Go itself. I can still try to make it more descriptive however.